### ISSUE TITLE
Default Invalid time offset Data In Fit Files to Local Timezone

### ISSUE THREAD

**Issue by fjih** [tcgoetz/GarminDB#252]
**Is your feature request related to a problem? Please describe.**
I have a great number of fit files that have invalid time offsets. The device is a Garmin Edge 530 cycling computer. The device does not have a configurable time. It instead pulls time data from either a gps signal or from a paired mobile phone. I suspect that if an activity is started before either aforementioned signals is acquired it results in an invalid time offset. I am seeing values of -4223384928.

As a result, GarminDB fails when it attempts to calculate the local timezone based on this invalid time offset.
The error is: `offset must be a timedelta strictly between -timedelta(hours=24) and timedelta(hours=24), not datetime.timedelta(days=-48882, seconds=19812).`

**Describe the solution you'd like**
The current behavior of GarminDB is to attempt to get the time offset via a number of fields in the fit file if they exist and fall back to assuming the data is recorded in utc time. I think this behavior is reasonable.

The issue is what to do should the data be invalid. Currently the error is logged and the file is discarded. I also think this is reasonable. However, defaulting to a timezone may be a preferable solution. It would allow users to ingest the data is a way that is minimally altered instead of leaving it unavailable to them.

Any value of time offset outside of the expected range (-24h,24h) is unmistakeably invalid. I propose that an invalid time offset default to the local system timezone. I would also suggest that the default of assuming utc be changed to local system timezone. The local timezone is more likely to be correct than utc in most use cases.

I do not believe GarminDB should be in the business of handling all types of invalid data. That is clearly outside the scope of the project. I propose this change as I believe it to be a minimally invasive one that would unblock myself and hopefully others. Should that not be the case please feel free to close as won't fix, and I'll have to write a tool to adjust the time offsets in my files :)

**Comment by tcgoetz** [comment]
Can you get me a stack trace for this from the log file?

**Comment by fjih** [comment]
```
ERROR:/home/fjih/.local/lib/python3.10/site-packages/garmindb/fit_data.py:Failed to parse /home/fjih/GarminDb/FitFiles/Activities/11039912580_ACTIVITY.fit: offset must be a timedelta strictly between -timedelta(hours=24) and timedelta(hours=24), not datetime.timedelta(days=-48882, seconds=19872).
ERROR:root:Failed to parse /home/fjih/GarminDb/FitFiles/Activities/11039912580_ACTIVITY.fit: offset must be a timedelta strictly between -timedelta(hours=24) and timedelta(hours=24), not datetime.timedelta(days=-48882, seconds=19872). - Traceback (most recent call last):
  File "/home/fjih/.local/lib/python3.10/site-packages/garmindb/fit_data.py", line 51, in process_files
    fit_file = fitfile.file.File(file_name, self.measurement_system)
  File "/home/fjih/.local/lib/python3.10/site-packages/fitfile/file.py", line 47, in __init__
    self.__sumarize()
  File "/home/fjih/.local/lib/python3.10/site-packages/fitfile/file.py", line 113, in __sumarize
    self.local_tz = datetime.timezone(datetime.timedelta(seconds=self.utc_offset))
ValueError: offset must be a timedelta strictly between -timedelta(hours=24) and timedelta(hours=24), not datetime.timedelta(days=-48882, seconds=19872).
```

**Comment by tcgoetz** [comment]
Unfortunately, it's catching and handling the exception to far up. Can you share one of these files for me to debug with?

**Comment by fjih** [comment]
https://www.dropbox.com/scl/fi/erojc301mmjlgp33qd7h5/activity_edit.fit?rlkey=r7aogwzbdob7n2csjr9xjprxr&st=esks5njm&dl=0

Sorry for the delay. I was cleaning the file of location data.

Maybe I should have posted this in the fitfile project instead.
The code is at fitfile/file.py lines 106 and 113.

On 106 it pulls the invalid time_offset from the file.
On 113 is fails trying to create a datetime.timezone from the invalid offset

**Comment by tcgoetz** [comment]
Fix pushed to develop. Can you test from the develop branch?

**Comment by fjih** [comment]
Thanks Tom. Develop branch is working for me.
